module/x/peggy/keeper/attestation.go

```diff
@@ -102,7 +77,8 @@ func (k Keeper) tryAttestation(ctx sdk.Context, att *types.Attestation, claim ty
 			attestationPower = attestationPower.Add(sdk.NewInt(validatorPower))
 			// If the power of all the validators that have voted on the attestation is higher or equal to the threshold,
 			// process the attestation, set Observed to true, and break
-			if attestationPower.GTE(requiredPower) {
+			lastEventNonce := k.GetLastObservedEventNonce(ctx)
+			if attestationPower.GTE(requiredPower) && claim.GetEventNonce() == uint64(lastEventNonce)+1 {
```

Comment:
@jtremback  this is the biggest change I made while getting things working.

Since the endblocker loop exits when the first attestation is not observed this simply means out of order attestations don't become observed and the loop exits then. It also makes any panic or out of order execution impossible. So I think it's good.